components/FilteredTextArea.tsx

```diff
@@ -15,16 +15,41 @@ export default function FilteredTextArea(props: FilteredTextAreaProps) {
 
   const handleChange = useCallback(
     (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+      e.currentTarget.value = [
+        ...chunks(
+          [
+            ...chunks(
+              e.currentTarget.value.replace(/[^a-z0-9]/gi, "").split(""),
+              2
+            ),
+          ].map((pair) => pair.join("")),
+          6
```

Comment:
Do I understand this correctly that it will force a new line after 6 digits? I've seen the code matrix size being anywhere from 4x4 to 7x7 (not sure if even 8x8).

Reply:
That's a good point. Anything we can do about it? Maybe we could be smart and dynamically adjust the matrix size according to a `sqrt(items)` formula.

Reply:
The least intrusive or confusing solution IMO is to just let the user add the newlines; to be out of the user's way. Maybe could just limit the automagic to add a space automatically after a digit?

Reply:
Does copypasting work, by the way?